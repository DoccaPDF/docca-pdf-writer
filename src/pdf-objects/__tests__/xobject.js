import XObject from '../xobject';

const xobject = XObject({
  id: 1,
  Subtype: 'Image',
  Filter: 'FlateDecode',
  BitsPerComponent: 8,
  Width: 100,
  Height: 20,
  ColorSpace: 'DeviceRGB',
  SMask: { id: 2 },
  data: 'fakeImageDate'
});

describe('pdf-objects', () => {
  describe('xobject', () => {
    it('creates a new object', () => {
      expect(xobject).toMatchSnapshot();
    });

    it('flattens keys values', () => {
      expect(xobject.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      expect(xobject.toPDF()).toMatchSnapshot();
    });
  });
});
