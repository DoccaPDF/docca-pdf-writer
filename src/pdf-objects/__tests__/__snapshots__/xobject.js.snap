// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects xobject creates a new object 1`] = `
Object {
  "BitsPerComponent": 8,
  "ColorSpace": "DeviceRGB",
  "Filter": "FlateDecode",
  "Height": 20,
  "SMask": Object {
    "id": 2,
  },
  "Subtype": "Image",
  "Width": 100,
  "data": "fakeImageDate",
  "id": 1,
}
`;

exports[`pdf-objects xobject flattens keys values 1`] = `
Object {
  "BitsPerComponent": 8,
  "ColorSpace": "/DeviceRGB",
  "DecodeParms": undefined,
  "Filter": "/FlateDecode",
  "Height": 20,
  "Length": 14,
  "Mask": undefined,
  "SMask": "2 0 R",
  "Subtype": "/Image",
  "Type": "/XObject",
  "Width": 100,
}
`;

exports[`pdf-objects xobject returns a PDF object 1`] = `
Object {
  "data": Array [
    49,
    32,
    48,
    32,
    111,
    98,
    106,
    10,
    60,
    60,
    10,
    47,
    84,
    121,
    112,
    101,
    32,
    47,
    88,
    79,
    98,
    106,
    101,
    99,
    116,
    10,
    47,
    83,
    117,
    98,
    116,
    121,
    112,
    101,
    32,
    47,
    73,
    109,
    97,
    103,
    101,
    10,
    47,
    76,
    101,
    110,
    103,
    116,
    104,
    32,
    49,
    52,
    10,
    47,
    66,
    105,
    116,
    115,
    80,
    101,
    114,
    67,
    111,
    109,
    112,
    111,
    110,
    101,
    110,
    116,
    32,
    56,
    10,
    47,
    67,
    111,
    108,
    111,
    114,
    83,
    112,
    97,
    99,
    101,
    32,
    47,
    68,
    101,
    118,
    105,
    99,
    101,
    82,
    71,
    66,
    10,
    47,
    87,
    105,
    100,
    116,
    104,
    32,
    49,
    48,
    48,
    10,
    47,
    72,
    101,
    105,
    103,
    104,
    116,
    32,
    50,
    48,
    10,
    47,
    70,
    105,
    108,
    116,
    101,
    114,
    32,
    47,
    70,
    108,
    97,
    116,
    101,
    68,
    101,
    99,
    111,
    100,
    101,
    10,
    47,
    83,
    77,
    97,
    115,
    107,
    32,
    50,
    32,
    48,
    32,
    82,
    10,
    62,
    62,
    10,
    115,
    116,
    114,
    101,
    97,
    109,
    10,
    102,
    97,
    107,
    101,
    73,
    109,
    97,
    103,
    101,
    68,
    97,
    116,
    101,
    10,
    101,
    110,
    100,
    115,
    116,
    114,
    101,
    97,
    109,
    10,
    101,
    110,
    100,
    111,
    98,
    106,
    10,
  ],
  "type": "Buffer",
}
`;
